solver/src/naive_solver.rs

```diff
@@ -0,0 +1,144 @@
+mod two_order_settlement;
+
+use crate::settlement::Settlement;
+use model::{OrderKind, TokenPair, UserOrder};
+use primitive_types::U512;
+use std::{cmp::Ordering, collections::HashMap};
+
+pub fn settle(orders: impl Iterator<Item = UserOrder>) -> Option<Settlement> {
```

Comment:
Nice, I assume this will be the public interface that eventually becomes a trait so we can have multiple solvers?

Reply:
Something like this, don't know exactly yet. We will see when we integrate a second solver.